python_modules/dagster-test/dagster_test/components/complex_schema_asset.py

```diff
@@ -31,5 +32,5 @@ def dummy(context: AssetExecutionContext):
 
         defs = Definitions(assets=[dummy])
         for post_processor in self.asset_post_processors or []:
-            defs = post_processor(defs)
+            defs = post_processor(DefinitionsHandle(defs))._defs
```

Comment:
see `DefinitionsHandle` below for more context on this - basically this is a more limited handle that post-processors can use to map over assets - if the post processor needs to do any deeper surgery, it has to commit to loading (non-subset) defs